### PR TITLE
Rewrite extensions in Rust

### LINKED ISSUE #466: Rewrite extensions in Rust
- Use https://github.com/BrettMayson/arma-rs
- ApolloClient
  - Use [reqwest](https://docs.rs/reqwest) for REST

### PR DESCRIPTION
**Pull request by jonpas**
**When merged this pull request will:**
- Remove old extension
  - [x] Remove old tooling
  - [x] Remove old source ~(at the end)~
- Implement Apollo Client in Rust - close #466 
  - [x] Use https://github.com/BrettMayson/arma-rs for Arma (RV)
  - [x] Use [reqwest](https://docs.rs/reqwest) for REST
  - [x] Use [simplelog](https://docs.rs/simplelog) for logging to file
- Windows DLL tooling and instructions
  - [x] x64
  - [x] x86
- Interface with SQF
  - [x] Communication (#201)
  - [x] Split overflow data (`getAccessibleItemClasses`)
  - [x] Versioning (version check in SQF)
- Improvements
  - [x] ~Copy to root and name appropriately (Cargo Build Script)~
  - [x] ~Build for Windows on Windows using MSVC (smaller size)~
  - [x] `strip` binaries
  - [x] [Min-size binaries](https://github.com/johnthagen/min-sized-rust)
  - [x] CI

### CODE REVIEW

**Comment by JoramD0** [comment]
Reminder for myself: Change uses of `getPlayerInfo` to use "new" types

File: addons/apollo/functions/fnc_getPlayerInfo.sqf [hunk 1]
```diff
@@ -31,40 +31,14 @@ if (_playerUID isEqualTo "_SP_PLAYER_" || {_playerUID isEqualto "_SP_AI_"}) exit
     []
 };
 
-private _requestedInfo = [];
+private _loadData = "tac_apollo_client" callExtension format ["%1%2/%3", "loadPlayer", getPlayerUID _player, GVAR(isDebug)];
```
Comment (jonpas) [inline addons/apollo/functions/fnc_getPlayerInfo.sqf:34]:
```suggestion
private _loadData = "tac_apollo_client" callExtension format ["%1%2/%3", "loadPlayer", _playerUID, GVAR(isDebug)];
```

File: addons/armory/functions/fnc_getDataChronos.sqf [hunk 2]
```diff
@@ -27,41 +27,11 @@ if !(["tac_apollo"] call ACEFUNC(common,isModLoaded)) exitWith {
...
-                // Reset
-                if (_loadData == "next") then {
-                    _armoryData pushBack _entry;
-                    _entry = [];
-                };
-            };
-        };
-    };
-};
+private _armoryData = parseSimpleArray _loadData;
```
Comment (jonpas) [inline addons/armory/functions/fnc_getDataChronos.sqf:32]: Should probably be inside the `if`.

**Comment by jonpas** [comment]
Cross-compile on Linux for Windows works nicely.

x64:
```sh
rustup target add x86_64-pc-windows-gnu
cargo build --target x86_64-pc-windows-gnu
```

x86:
```sh
rustup target add i686-pc-windows-gnu
cargo build --target i686-pc-windows-gnu
```

**Comment by jonpas** [comment]
> Cross-compile on Linux for Windows works nicely.

`jni-rs` has shown natively building on Windows produces much smaller binaries. Can do that with CI for ease.

**Pull request merged**